lib/gollum/app.rb

```diff
@@ -590,7 +590,7 @@ class App < Sinatra::Base
     end
 
     get '/*' do
```

Comment:
I don't have all the details at the moment, but would it be more straightforward to check if `params.has_key?("raw")` here in this route, and, if so, call a new method `show_raw_contents(params[:splat].first)`?

Reply:
It makes sense to me to use `show_file` because `raw` essentially means "retrieve that wiki page and skip rendering" - basically just "give me the file". If you accept that, `show_page_or_file` looks like the perfect place to use the existing call to `show_file` and also to only give `?raw` an effect when you know you're serving an actual wiki page (and not a redirect to another page or a creation view).

If you don't buy that, I'll defer to you, of course. :)

Reply:
Done, and see the other thread - I took a stab at the other approach in another branch, linked there.